pkg/apiserver/user/sqlauth/sqlauth.go

```diff
@@ -176,3 +174,21 @@ func checkDashboardPrivileges(grants []string, enableSEM bool) bool {
 	}
 	return true
 }
+
+func checkWriteablePriv(grants []string) bool {
+	privsMap := funk.Map(grants, func(priv string) (string, bool) {
+		return priv, true
+	}).(map[string]bool)
+	hasPriv := func(priv string) bool {
+		_, ok := privsMap[priv]
+		return ok
+	}
```

Comment:
How about returning grants as a map from the beginning? As you can see you have multiple places transforming from array to map.

Reply:
I have thought that, although changing grants to map in the method parameter reduces some duplicated logic, but the map type is not that reasonable for the 2 methods, because just the keys of the map are meaningful for them, we can feel that when we write unit tests for them.

But I only insist on this when they only are duplicated twice.

Reply:
If you want a set, maybe you can use `map[string]struct{}`. This is a common pattern to make a Set instead of a Map in golang.

Reply:
Done!